src/serializer/ResidualHeapSerializer.js

```diff
@@ -2047,6 +2049,10 @@ export class ResidualHeapSerializer {
   }
 
   _serializeEmptyValue(): BabelNodeExpression {
+    if (this.emptySerializesToEmptyString) {
+      invariant(this.realm.react.enabled, "emptySerializesToEmptyString is a React optimization");
+      return t.stringLiteral("");
```

Comment:
Rather than output the empty variable, instead we generate an empty string as the React serialization expects this for cases where the value is a string.

Reply:
The special "empty value" is used as a marker value as part of comparisons. It should never leak through as the actual value of anything that's exposed. For that reason, I don't understand how this can fix anything?

Also, the identity if that "empty value" matters. Replacing it in some contexts with something else (that isn't even unique), seems highly dubious to me. Why is this correct?

Reply:
I've removed this form the PR, but now the tests fail as `__empty` is no longer an empty string but an empty object which gets serialized out and is invalid as a return value for a React render.

Reply:
I have a different approach that removes holes from the arrays in the reconciler so we shouldn't hit the same issue but the still overall problem with `__empty` exists, see the issue above.